### PR TITLE
[RHCLOUD-34875] Pendo add widgets tracking

### PR DESCRIPTION
**Pull request by tahmidefaz**
### Description

Pendo analytics tracking for widgets move, add and remove. Uses the `analytics` functions provided by `useChrome()`. Uses the `widgetType` as the identifier for the Pendo events.

[RHCLOUD-34875](https://issues.redhat.com/browse/RHCLOUD-34875)

---

#### Before:
NA

#### After:
NA

---

### Checklist ☑️
- [x] PR only fixes one issue or story 
- [x] Change reviewed for extraneous code 
- [x] UI best practices adhered to 
- [x] Commits squashed and meaningfully named 
- [x] All PR checks pass locally (build, lint, test, E2E)

##
- [ ] _(Optional) QE: Needs QE attention (OUIA changed, perceived impact to tests, no test coverage)_
- [ ] _(Optional) QE: Has been mentioned_
- [ ] _(Optional) UX: Needs UX attention (end user UX modified, missing designs)_
- [ ] _(Optional) UX: Has been mentioned_
##

### CODE REVIEW

File: src/Components/DnDLayout/GridTile.tsx [hunk 1]
```diff
@@ -153,7 +158,10 @@ const GridTile = ({ widgetType, isDragging, setIsDragging, setWidgetAttribute, w
       </Tooltip>
       <Tooltip aria-label="Move widget" content={<p>{widgetConfig.static ? 'Widget locked' : 'Move'}</p>}>
         <Icon
-          onMouseDown={() => setIsDragging(true)}
+          onMouseDown={() => {
+            setIsDragging(true);
+            analytics.track('widget-move', { widgetTitle });
+          }}
```
Comment (tahmidefaz) [inline src/Components/DnDLayout/GridTile.tsx:164]: @florkbr It seems to trigger this event every time the user clicks on the move icon on the widgets. We might need to find something else for the analytics to listen to?

**Review by florkbr — CHANGES_REQUESTED** [review]
Thanks Tahmid! 2 small requests:
1. move the add widget track event to `GridLayout.tsx > onDrop` (inside the if conditional, we can use the `data` param/string there in the track as it is the string we lookup in widgetMapping)
2. we should use `widgetType` in our analytics calls in GridTile.tsx, not `widgetTitle` (as we use widgetType to lookup the widget in the widget mapping)

File: src/Components/DnDLayout/GridTile.tsx (hunk 1 above)
Comment (florkbr) [reply src/Components/DnDLayout/GridTile.tsx:164]: @tahmidefaz I think for now (based on the Jira ticket) we just want to know if a widget it being moved and if so which ones. I think putting the listener on the drag handle works for now, and then we will need to investigate more in the future around:
1. Which widget appears first in the layout?
2. When a widget is moved, where is it moved from/to?

**Comment by tahmidefaz** [comment]
@florkbr all the requested changes should be addressed now. Please let me know if more changes are needed. Thank you!

**Comment by florkbr** [comment]
@tahmidefaz it looks like the konflux enterprise contract is failing. I'm thinking it's because we need this: https://github.com/RedHatInsights/widget-layout/pull/103

**Comment by florkbr** [comment]
Let me open a pull request to confirm it's not your changes

**Comment by florkbr** [comment]
Looks like https://github.com/RedHatInsights/widget-layout/pull/108 (readme change) failed for the same reason - unrelated to this PR. I'll merge this for now and prioritize getting the konflux update in.

**Review by florkbr — APPROVED** [review]

**Pull request merged**